source/s3_client.c

```diff
@@ -134,6 +161,18 @@ void aws_s3_set_dns_ttl(size_t ttl) {
     s_dns_host_address_ttl_seconds = ttl;
 }
 
+static int s_s3_request_waiting_for_mem_priority_queue_pred(const void *a, const void *b) {
+    const struct aws_s3_request *const *request_a = a;
+    AWS_PRECONDITION(request_a);
+    AWS_PRECONDITION(*request_a);
+
+    const struct aws_s3_request *const *request_b = b;
+    AWS_PRECONDITION(request_b);
+    AWS_PRECONDITION(*request_b);
+
+    return (*request_a)->num_times_tried_buffer_acquire > (*request_b)->num_times_tried_buffer_acquire;
```

Comment:
can we prioritize certain requests like CancelMultipartUpload? Isn't there already some high-priority-request concept? AWS_S3_REQUEST_FLAG_ALWAYS_SEND?